### DISCUSSION TITLE
Save option for modifications

### DISCUSSION THREAD

**Discussion by enzu4l** [mriscoc/Ender3V2S1#289] [Ideas]
Hi there,
Cause I’m not into programming I can’t imagine how hard it would be to add an option where we could save values for our mods we add to the Ender 3v2.
Right now I’m with an offset for CLtouch, different position of extruder and the bed. Every time I update the firmware, I have to enter everything again. Restore doesn’t work. Or am I just dump?
Thanks again for all your effort, making a good printer awesome.

**Comment by ghost** [comment]
It would really be nice if there was a possibility to export the settings to the sdcard and after updating we can go back

**Reply by enzu4l** [reply]
This is what I meant. Cheers!

**Comment by mriscoc** [comment]
You can write a config.gco file and apply it after updating the firmware, for example:

```gcode
; Professional Firmware Configuration File
;=====================================================
C10                    ; Mark as a configuration file
M117 Applying configuration
;-----------------------------------------------------
M256 B150              ; Set Brightness 
M413 S0                ; Disable Powerloss recovery
M603 L200.00 U200.00   ; Configure Filament Change
C851 S800 M0           ; Probe feedrate and disable multiple probing
C29 L25.00 R195.00 F25.00 B195.00 N5 T50 ; Set Mesh insets, size of 5x5 and bed at 50°C
C100 X0 Y0             ; Set XY Min position to 0
C101 X230 Y230 Z240    ; Set XYZ Max position
C102 X220 Y220         ; Set a bed size of 220x220
C412 M0                ; Filament sensor active mode: LOW
;-----------------------------------------------------
G4 S1                       ; Wait a second
M300 P200                   ; Beep
M117 Configuration Applied
```

For set probe offsets you can add the g-code M851 to the configuration file.
